### ISSUE TITLE
Issue with Invalid Access Token scenario

### ISSUE THREAD

**Issue by FLieshout** [camaraproject/SimSwap#155]
**Problem description**
When an invalid token is specified with, for example, the wrong scope is given, I expect a different type of error.

```FEATURE
@retrieve_sim_swap_date_401.3_invalid_access_token
  Scenario: Invalid access token
    Given the header "Authorization" is set to an invalid access token
    And the request body is set to a valid request body
    When the request "retrieveSimSwapDate" is sent
    Then the response status code is 401
    And the response property "$.status" is 401
    And the response property "$.code" is "UNAUTHENTICATED"
    And the response property "$.message" contains a user friendly text
```

**Expected behavior**
In the [API specs](https://github.com/camaraproject/SimSwap/blob/r1.2/code/API_definitions/sim-swap.yaml) this is described: _Permission denied. OAuth2 token access does not have the required scope or when the user fails operational security._ 

```JSON
{
  "status": 403,
  "code": "PERMISSION_DENIED",
  "message": "Client does not have sufficient permissions to perform this action."
}
```

**Alternative solution**:

```FEATURE
@retrieve_sim_swap_date_401.3_invalid_access_token
  Scenario: Invalid access token
    Given the header "Authorization" is set to an invalid access token
    And the request body is set to a valid request body
    When the request "retrieveSimSwapDate" is sent
    Then the response status code is 403
    And the response property "$.status" is 403
    And the response property "$.code" is "PERMISSION_DENIED"
    And the response property "$.message" contains a user friendly text
```

**Additional context**
Or is the 'invalid access token' a different error than the one I described in the problem description?

- Same test scenario is described in the /check API.

**Comment by fernandopradocabrillo** [comment]
Hi @FLieshout 
Those are indeed two different scenarios.
401 UNAUTHENTICATED -> when the credentials are invalid for different reasons i.e the token has expired or the value provided is invalid in format, etc
403 PERMISSION_DENIED -> the token is valid but the client doesn't have permission to access the resource, i.e the token is lacking the required scope

In other test plans for other CAMARA APIs like number verification, the 403 is considered but, in general, CAMARA test plans do not cover specific authentication errors, just the basic 401.

cc: @bigludo7

**Comment by bigludo7** [comment]
@fernandopradocabrillo This is probably an area of improvement as I'm not sure we have the same rules in every APIs. I checked 4 of the 5 stable APIs (OTP works in 2-legs so not relevant).

I saw some discrepancies... wondering if we should no ask the ICM team to provide guidance? 

In location verification
We have 401  for no authorization header, expired access token and invalid access token. 
403 for device token mismatch

In number verification verify
401 for expired access token
403 for missing scope, no phone associated with the access token and no network authentification (this latest is specific for this API)

in Sim swap check
401 for no authorization header, expired access token and invalid access token. 
403 for for device token mismatch but also phone number cannot be deducted from access token

In Simple edge Discovery
401 - no scenario
403  invalid access token
